### ISSUE TITLE
How to handle out-of-class refs?

### ISSUE THREAD

**Issue by oliverwoodings** [QubitProducts/react-test-tree#36]
@jhollingworth @giuse88 could do with some input on this one.

Right now, testTree does not support either of the following use cases (they are functionally identical):

```jsx
// MyComponent renders `this.props.children`
let tree = testTree(
  <MyComponent>
    <input ref="textInput" />
  </MyComponent>
)
// I want to access textInput but I can't
tree.textInput // -> undefined
```

```jsx
// This time no outer component, but same principle
let tree = testTree(
  <div>
    <input ref="textInput" />
  </div>
  // I want to access textInput but I can't
  tree.textInput // -> undefined
)
```

The reason we can't access `textInput` is because refs belong to the `render` function they are directly returned from. Internally we wrap all calls in a HOC (for many reasons), so in this case `textInput` ends up being rendered by testTree's HOC.

This isn't an easy problem to solve, since right now we return the tree starting at the top-level component passed to testTree (i.e. `MyComponent`). Changing to always return the tree starting at the HOC doesn't fix the issue because you then require the user to always put a ref on the top-level component (unnecessary in the majority of cases).

I'm wondering whether we should add a flag to the config options called `wrap` that instructs testTree to return the tree starting at the HOC, like so:

```jsx
let tree = testTree(
  <MyComponent ref="myComponent">
    <input ref="textInput" />
  </MyComponent>
, { wrap: true })
// textInput is now defined
tree.textInput
// so is myComponent
tree.myComponent
```

Thoughts?

**Comment by austinpray** [comment]
I don't have any issue with adding the `wrap: true`. It's backwards compatible too.

**Comment by oliverwoodings** [comment]
Closed by #31
